api/src/database/helpers/schema/types.ts

```diff
@@ -139,4 +139,11 @@ export abstract class SchemaHelper extends DatabaseHelper {
 	formatUUID(uuid: string): string {
 		return uuid; // no-op by default
 	}
+
+	/**
+	 * @returns Size of the database in bytes
+	 */
+	async getDatabaseSize(): Promise<number> {
+		return 0;
```

Comment:
Should we return `null` here as well as for failing `getDatabaseSize` calls (`catch` clause) in the dialects implementation so we can identify these cases?

Reply:
Changes pushed via
https://github.com/directus/directus/pull/22337/commits/3b45bf9dc974549590e537c558d135900661a7ab.

> For the newly introduced data (db size, and especially fields count) do we expect them to perform well

The new queries should be fairly light IMO 🤔 

> Just because it might be good to know (and maybe add inline docs): Do (some of) the DB size queries require some special DB permissions?

There's potential for permission issue on certain DBs, but it should not affect anything operationally as a null is return if it errors.